File: forgerock-auth/src/test/java/org/forgerock/android/auth/callback/SelectIdPCallbackTest.java
/*
 * Copyright (c) 2021 ForgeRock. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

package org.forgerock.android.auth.callback;

import org.json.JSONException;
import org.json.JSONObject;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;

import static org.assertj.core.api.Assertions.assertThat;
import static org.junit.Assert.assertEquals;

@RunWith(RobolectricTestRunner.class)
public class SelectIdPCallbackTest {

    @Test
    public void basicTest() throws JSONException {
        JSONObject raw = new JSONObject("{\n" +
                "            \"type\": \"SelectIdPCallback\",\n" +
                "            \"output\": [\n" +
                "                {\n" +
                "                    \"name\": \"providers\",\n" +
                "                    \"value\": [\n" +
                "                        {\n" +
                "                            \"provider\": \"google\",\n" +
                "                            \"uiConfig\": {\n" +
                "                                \"buttonImage\": \"images/g-logo.png\",\n" +
                "                                \"buttonCustomStyle\": \"background-color: #fff; color: #757575; border-color: #ddd;\",\n" +
                "                                \"abc\": \"test\",\n" +
                "                                \"buttonClass\": \"\",\n" +
                "                                \"buttonDisplayName\": \"Google\",\n" +
                "                                \"buttonCustomStyleHover\": \"color: #6d6d6d; background-color: #eee; border-color: #ccc;\",\n" +
                "                                \"iconClass\": \"fa-google\",\n" +
                "                                \"userinfo\": \"https://www.googleapis.com/oauth2/v3/userinfo\",\n" +
                "                                \"iconFontColor\": \"white\",\n" +
                "                                \"iconBackground\": \"#4184f3\"\n" +
                "                            }\n" +
                "                        },\n" +
                "                        {\n" +
                "                            \"provider\": \"facebook\",\n" +
                "                            \"uiConfig\": {\n" +
                "                                \"buttonImage\": \"\",\n" +
                "                                \"buttonCustomStyle\": \"background-color: #3b5998;border-color: #3b5998; color: white;\",\n" +
                "                                \"buttonClass\": \"fa-facebook-official\",\n" +
                "                                \"buttonDisplayName\": \"Facebook\",\n" +
                "                                \"buttonCustomStyleHover\": \"background-color: #334b7d;border-color: #334b7d; color: white;\",\n" +
                "                                \"iconFontColor\": \"white\",\n" +
                "                                \"iconClass\": \"fa-facebook\",\n" +
                "                                \"iconBackground\": \"#3b5998\"\n" +
                "                            }\n" +
                "                        },\n" +
                "                        {\n" +
                "                            \"provider\": \"localAuthentication\"\n" +
                "                        }\n" +
                "                    ]\n" +
                "                },\n" +
                "                {\n" +
                "                    \"name\": \"value\",\n" +
                "                    \"value\": \"\"\n" +
                "                }\n" +
                "            ],\n" +
                "            \"input\": [\n" +
                "                {\n" +
                "                    \"name\": \"IDToken1\",\n" +
                "                    \"value\": \"\"\n" +
                "                }\n" +
                "            ]\n" +
                "        }");
        SelectIdPCallback callback = new SelectIdPCallback(raw, 0);
        assertThat(callback.getProviders()).hasSize(3);
        assertThat(callback.getProviders().get(0).getProvider()).isEqualTo("google");
        assertThat(new JSONObject(callback.getProviders().get(0).getUiConfig()).getString("buttonDisplayName"))
                .isEqualTo("Google");
        assertThat(callback.getProviders().get(1).getProvider()).isEqualTo("facebook");
        assertThat(new JSONObject(callback.getProviders().get(1).getUiConfig()).getString("buttonDisplayName"))
                .isEqualTo("Facebook");
        assertThat(callback.getProviders().get(2).getProvider()).isEqualTo("localAuthentication");

        callback.setValue("google");
        assertEquals(((JSONObject)callback.getContentAsJson().getJSONArray("input").get(0)).getString("value"),
                "google");

        assertEquals(0, callback.get_id());

    }
}

File: forgerock-auth/src/test/java/org/forgerock/android/auth/callback/SelectIdPCallbackTest.kt
/*
 * Copyright (c) 2023 ForgeRock. All rights reserved.
 *
 *  This software may be modified and distributed under the terms
 *  of the MIT license. See the LICENSE file for details.
 */
package org.forgerock.android.auth.callback

import androidx.test.ext.junit.runners.AndroidJUnit4
import org.assertj.core.api.Assertions
import org.json.JSONException
import org.json.JSONObject
import org.junit.Assert
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
class SelectIdPCallbackTest {
    @Test
    @Throws(JSONException::class)
    fun basicTest() {
        val raw = JSONObject("""{
            "type": "SelectIdPCallback",
            "output": [
                {
                    "name": "providers",
                    "value": [
                        {
                            "provider": "google",
                            "uiConfig": {
                                "buttonImage": "images/g-logo.png",
                                "buttonCustomStyle": "background-color: #fff; color: #757575; border-color: #ddd;",
                                "abc": "test",
                                "buttonClass": "",
                                "buttonDisplayName": "Google",
                                "buttonCustomStyleHover": "color: #6d6d6d; background-color: #eee; border-color: #ccc;",
                                "iconClass": "fa-google",
                                "userinfo": "https://www.googleapis.com/oauth2/v3/userinfo",
                                "iconFontColor": "white",
                                "iconBackground": "#4184f3"
                            }
                        },
                        {
                            "provider": "facebook",
                            "uiConfig": {
                                "buttonImage": "",
                                "buttonCustomStyle": "background-color: #3b5998;border-color: #3b5998; color: white;",
                                "buttonClass": "fa-facebook-official",
                                "buttonDisplayName": "Facebook",
                                "buttonCustomStyleHover": "background-color: #334b7d;border-color: #334b7d; color: white;",
                                "iconFontColor": "white",
                                "iconClass": "fa-facebook",
                                "iconBackground": "#3b5998"
                            }
                        },
                        {
                            "provider": "localAuthentication"
                        }
                    ]
                },
                {
                    "name": "value",
                    "value": ""
                }
            ],
            "input": [
                {
                    "name": "IDToken1",
                    "value": ""
                }
            ]
        }""")
        val callback = SelectIdPCallback(raw, 0)
        Assertions.assertThat(callback.providers).hasSize(3)
        Assertions.assertThat(callback.providers[0].provider).isEqualTo("google")
        Assertions.assertThat(JSONObject(callback.providers[0].uiConfig).getString("buttonDisplayName"))
            .isEqualTo("Google")
        Assertions.assertThat(callback.providers[1].provider).isEqualTo("facebook")
        Assertions.assertThat(JSONObject(callback.providers[1].uiConfig).getString("buttonDisplayName"))
            .isEqualTo("Facebook")
        Assertions.assertThat(callback.providers[2].provider).isEqualTo("localAuthentication")
        callback.setValue("google")
        Assert.assertEquals((callback.contentAsJson.getJSONArray("input")[0] as JSONObject).getString(
            "value"),
            "google")
        Assert.assertEquals(0, callback.get_id().toLong())
    }
}